### ISSUE TITLE
S3 Backups Broken

### ISSUE THREAD

**Issue by patrickstump** [tiredofit/docker-openldap#35]
There are several issues with trying to use S3 Backups.

## Environment
Using the docker-compose file provided

### Setting Backups to S3
In order to set the backups to S3, the env variable `BACKUP_TYPE` needs to be set to `S3`, not `BACKUP_LOCATION`.
This can be seen at [/etc/services.available/20-openldap-backup/run line 17](https://github.com/tiredofit/docker-openldap/blob/1fd264d7caf8a136ece9dcf4ea50c6cf884257ec/install/etc/services.available/20-openldap-backup/run#L17)

### S3 Environment variables are incorrect
1. The S3 variable do not want `BACKUP_` in front of them, as can be seen at starting at [/etc/services.available/20-openldap-backup/run line 128] (https://github.com/tiredofit/docker-openldap/blob/1fd264d7caf8a136ece9dcf4ea50c6cf884257ec/install/etc/services.available/20-openldap-backup/run#L128)
2. The `S3_HOSTNAME` is not correct, the script wants `S3_HOST`

### Missing libressl

Don't know if it is related or not
```
++ libressl md5 -binary
/etc/services.available/20-openldap-backup/run: line 149: libressl: command not found
```

**Comment by tiredofit** [comment]
Thanks for this, will review and push out a fix.

**Comment by tiredofit** [comment]
Just looking at this in detail: - My apologies, I copied this from by `tiredofit/db-backup` image and didn't do proper QA. 

- Setting Backup Type to S3 (line 17) I am not following:
My line is: 
`if [ "$BACKUP_TYPE" = "S3" ] || [ "$BACKUP_TYPE" = "s3" ] || [ "$BACKUP_TYPE" = "MINIO" ] || [ "$BACKUP_TYPE" = "minio" ] ; then`
 
Which doesn't anywhere state `BACKUP_LOCATION` - Can I Get clarity?

- S3 Variables are not correct - Again that is a `BACKUP_TYPE` Situation (You would set variable `BACKUP_TYPE=S3`) to activate.
- Correct on the S3_HOSTNAME - I've updated the documentation to state `S3_HOST` on the next commit.

- LibreSSL issue- Nice catch. We're using openssl from this image, and again I have to apologize for not doing proper QA on the S3 component, the next push will contain this fix.

**Comment by tiredofit** [comment]
Initial Change pushed as `tiredofit/openldap:7.1.11`

**Comment by patrickstump** [comment]
Yeah sorry if the comment about `BACKUP_LOCATION` didn't make sense... end of the day and I was a bit tired.

In the Environment Variables section of the read me, it says that `BACKUP_LOCATION` specifies either filesystem or or s3.
I was attempting to say that the code was looking for `BACKUP_TYPE` and not using `BACKUP_LOCATION`.

**Comment by patrickstump** [comment]
We're using your image `openldap-fusiondirectory`, can you trigger a build so it comes through on that one?

**Comment by tiredofit** [comment]
Sure, I've just rebuilt it (`:latest`).

**Comment by patrickstump** [comment]
that seems

**Comment by patrickstump** [comment]
sorry. i'm use to closing issues at my work :)...

You probably meant for it to be this way but `/usr/loca/bin/backup-now` is not executable. you have to `bash ./backup-now`.
just an fyi

**Comment by tiredofit** [comment]
Darnit, thats my permissiosn on my workstation that got reset after a rebuild. Apologies on being very messy. I will resolve shortly.
